### PR TITLE
fix MasterBaseTaskExecThread submit method bug

### PR DESCRIPTION
**Pull request by Technoboy-**

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java [hunk 1]
```diff
@@ -114,37 +114,37 @@ protected TaskInstance submit(){
...
                 }
-                if(taskDBFlag && taskQueueFlag){
+                if(submitDB && submitQueue){
                     return task;
                 }
-                if(!taskDBFlag){
+                if(!submitDB){
                     logger.error("task commit to db failed , task has already retry {} times, please check the database", retryTimes);
-                }else if(!taskQueueFlag){
+                }else if(!submitDB){
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterBaseTaskExecThread.java]: is submitQueue?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterTaskExecThread.java [hunk 2]
```diff
@@ -74,6 +74,9 @@ public TaskInstance getTaskInstance(){
     public Boolean submitWaitComplete() {
         Boolean result = false;
         this.taskInstance = submit();
+        if(this.taskInstance == null){
+            return result;
```
Comment (lenboo) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterTaskExecThread.java]: should this return be checked whether the process instance is can be continue normally?
i am not sure about this.

Comment (Technoboy-) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterTaskExecThread.java]: @qiaozhanwei  how about your idea

Comment (Technoboy-) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterTaskExecThread.java]: Should we send alert here if submit failed?

Comment (qiaozhanwei) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterTaskExecThread.java]: 1，Uncertain return, it is recommended to debug
2，submit failed should be logs and mail

Comment (Technoboy-) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterTaskExecThread.java]: yes, I debug for submit fail test, process will end and be a failure state.
it is satisfied the result.

**Pull request merged**